src/Microsoft.Azure.ServiceBus/MessagingEventSource.cs

```diff
@@ -1150,6 +1150,21 @@ void ExceptionReceivedHandlerThrewException(string exception)
             WriteEvent(99, exception);
         }
 
+        [NonEvent]
+        public void LinkStateLost(string clientId, string receiveLinkName, AmqpObjectState receiveLinkState, bool isSessionReceiver, Exception exception)
+        {
+            if (this.IsEnabled())
+            {
+                this.LinkStateLost(clientId, receiveLinkName, receiveLinkState.ToString(), isSessionReceiver, exception.ToString());
```

Comment:
Maybe. But if you are just looking at the logs, having the stack trace around could help in faster debugging. Extra logging might not a big issue anyway in the case of exceptions. Leaving it as-is